pkg/clusteragent/externalmetrics/metrics_retriever.go

```diff
@@ -63,7 +69,45 @@ func (mr *MetricsRetriever) Run(stopCh <-chan struct{}) {
 
 func (mr *MetricsRetriever) retrieveMetricsValues() {
 	// We only update active DatadogMetrics
-	datadogMetrics := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool { return datadogMetric.Active })
+	// We split metrics in two slices, those with errors and those without.
+	// Query first slice one by one, other as batch.
+	datadogMetrics := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool {
+		return datadogMetric.Active && datadogMetric.Error == nil
+	})
+
+	// Do all errors warrant separate query?
+	datadogMetricsErr := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool {
+		return datadogMetric.Active && datadogMetric.Error != nil
+	})
+
+	// First split then query because store state is shared and query mutates it
+	mr.retrieveMetricsValuesSlice(datadogMetrics)
+	for _, metrics := range datadogMetricsErr {
```

Comment:
can be good to add a empty line between these 2 lines
and a comment
```suggestion
	mr.retrieveMetricsValuesSlice(datadogMetrics)
	
	// Now test each metric query separately respecting its backoff retry duration elapse value. 
	for _, metrics := range datadogMetricsErr {
```